gap/congruences/conglatt.gi

```diff
@@ -159,41 +126,63 @@ SEMIGROUPS.LatticeOfXCongruences := function(S, type_string, record)
     fi;
   od;
 
-  congs := ShallowCopy(congs1);
-  # We now have all 1-generated congs, which must include all the minimal
-  # congs.  We can return if necessary.
-  if IsBound(record.minimal) and record.minimal = true then
-    # Find all the minimal congs (those with no children)
-    congs := congs{Positions(children, [])};
-    # Note: we don't include the trivial cong
-    # Set the MinimalXCongruencesOfSemigroup attribute
-    if not IsBound(record.restriction) then
-      set_func := EvalString(Concatenation("SetMinimal",
-                                           type_string,
-                                           "CongruencesOfSemigroup"));
-      set_func(S, congs);
-    fi;
-    # Minimal congs cannot contain each other
-    children := ListWithIdenticalEntries(Length(congs), []);
-    lattice := Objectify(NewType(FamilyObj(children),
-                                 IsCongruenceLattice),
-                         [children, congs]);
-    SEMIGROUPS.OptionsRec(S).report := report;
-    return lattice;
-  elif IsBound(record.1gen) and record.1gen = true then
-    # Return the lattice, but don't set any attributes
-    lattice := Objectify(NewType(FamilyObj(children),
-                                 IsCongruenceLattice),
-                         [children, congs]);
-    SEMIGROUPS.OptionsRec(S).report := report;
-    return lattice;
+  SEMIGROUPS.OptionsRec(S).report := report;
+
+  # We are done: make the object and return
+  po := Digraph(children);
+  SetInNeighbours(po, parents);
+  poset := Objectify(NewType(FamilyObj(children), IsCongruencePoset),
+                     rec(po := po));
+  SetCongruencesOfPoset(poset, congs);
+  SetRange(poset, S);
```

Comment:
Is `Range` really the appropriate attribute to store the semigroup in? Shouldn't it be something like `UnderlyingSemigroupOfPosetOfCongruences` or something preferably shorter?